yarn-project/prover-client/src/prover-pool/circuit-prover-agent.ts

```diff
@@ -0,0 +1,102 @@
+import { makeEmptyProof } from '@aztec/circuits.js';
+import { createDebugLogger } from '@aztec/foundation/log';
+import { RunningPromise } from '@aztec/foundation/running-promise';
+import { elapsed } from '@aztec/foundation/timer';
+
+import { type CircuitProver } from '../prover/interface.js';
+import { type ProvingAgent } from './prover-agent.js';
+import { type ProvingQueueConsumer } from './proving-queue.js';
+import { type ProvingRequest, type ProvingRequestResult, ProvingRequestType } from './proving-request.js';
+
+export class CircuitProverAgent implements ProvingAgent {
+  private runningPromise?: RunningPromise;
+
+  constructor(
+    /** The prover implementation to defer jobs to */
+    private prover: CircuitProver,
+    /** How long to wait between jobs */
+    private intervalMs = 10,
+    /** A name for this agent (if there are multiple agents running) */
+    name = '',
+    private log = createDebugLogger('aztec:prover-client:prover-pool:agent' + (name ? `:${name}` : '')),
+  ) {}
+
+  start(queue: ProvingQueueConsumer): void {
+    this.runningPromise = new RunningPromise(async () => {
+      this.log.verbose('Getting proving jobs');
```

Comment:
The logs in this class should probably be `debug`. Going to get a lot of spam otherwise.

Reply:
I've removed them logs as I only used them during testing :grimacing: